Sources/Logging/Logging.swift

```diff
@@ -100,7 +100,28 @@ public enum LogLevel: Int {
     case error
 }
 
-public typealias LoggingMetadata = [String: String]
+// logging with String value, "before"
+//    measuring: 100 logs:
+//    // [some-id][2018-12-17 10:26:41 +0000][info][id=some-id trace=0-AF7651916CD43DD8448EB211C80319C-F067AA0BA902B7-0 per-data0=some-metadata additional=some-per-context-info] Hello 100
+//    7ms 655μs, 7ms 50μs, 6ms 155μs, 5ms 825μs, 21ms 72μs, 6ms 314μs, 6ms 96μs, 10ms 315μs, 5ms 653μs, 7ms 46μs,
+//    measuring: no 100 logs:
+//    1ms 223μs, 1ms 295μs, 1ms 260μs, 1ms 340μs, 1ms 386μs, 1ms 419μs, 1ms 525μs, 1ms 476μs, 1ms 482μs, 1ms 608μs,
+//
+// String:String, with didSet rendering
+//    measuring: 100 logs:
+//    // 2018-12-17 11:35:49 +0000 id=some-id additional=some-per-context-info per-data0=some-metadata trace=0-AF7651916CD43DD8448EB211C80319C-F067AA0BA902B7-0 Hello 100
+//    10ms 915μs, 6ms 531μs, 6ms 68μs, 5ms 735μs, 5ms 870μs, 9ms 493μs, 6ms 564μs, 7ms 400μs, 10ms 455μs, 7ms 205μs,
+//    measuring: no 10000 logs:
+//    1ms 643μs, 1ms 592μs, 1ms 671μs, 1ms 579μs, 1ms 602μs, 1ms 565μs, 1ms 560μs, 1ms 539μs, 1ms 510μs, 1ms 499μs,
+//
+// String:Any, avoiding didSet, rendering only when actually printing
+//    measuring: 100 logs:
+//    // [2018-12-17 12:04:45 +0000][info][additional=some-per-context-info trace=0-AF7651916CD43DD8448EB211C80319C-F067AA0BA902B7-0 per-data0=some-metadata id=some-id] Hello 100
+//    20ms 89μs, 6ms 877μs, 8ms 257μs, 10ms 628μs, 6ms 299μs, 5ms 718μs, 5ms 723μs, 5ms 362μs, 8ms 759μs, 6ms 155μs,
+//    measuring: no 10000 logs:
+//    675μs 82ns, 394μs 940ns, 393μs 986ns, 393μs 986ns, 393μs 986ns, 394μs 940ns, 395μs 59ns, 394μs 940ns, 395μs 59ns, 486μs 16ns,
```

Comment:
These are the results I'm trying to make nice; cases where we propagate those "expensive to render" things, but we never actually print so it does not need to be rendered into its string representation.

Note that when we DO print them, we may indeed print them a few times... but we could work around this with some caching of the "heavy reprs"? I'm more worried about the trace/debug logs not impacting performance when they're off, but causing rendering of the strings anyway, since the metadata object forcing them to, even if loglevel would mean they should not ever appear.